### PR TITLE
feat: add `swiftlint` builtin diagnostics and formatting

### PR DESCRIPTION
**Pull request by Ty3uK**
This PR adds support for swiftlint as diagnostics and formatting source.

Sorry for poor description, not sure how to properly describe this tiny feature 🙈

https://user-images.githubusercontent.com/2384454/207402202-e0f19862-bf65-4aab-97bb-67c857332549.mp4

In this video I use both `swiftlint` and `swiftformat` sources

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics/swiftlint.lua [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        url = "https://github.com/realm/SwiftLint",
+        description = "A tool to enforce Swift style and conventions.",
+    },
+    method = DIAGNOSTICS,
+    filetypes = { "swift" },
+    generator_opts = {
+        command = "swiftlint",
+        args = { "--reporter", "json", "--use-stdin", "--quiet" },
+        to_stdin = true,
+        format = "json_raw",
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/swiftlint.lua:36]: We generally only want to use `json_raw` if you are specifically handling errors, which doesn't seem to be the case. Was there a specific reason you chose this over `json`?

File: lua/null-ls/builtins/diagnostics/swiftlint.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        use_cache = true,
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/swiftlint.lua:37]: This option only has an effect if you have multiple sources using the same output, so I don't think it's necessary here.

File: lua/null-ls/builtins/diagnostics/swiftlint.lua (hunk 1 above)
Comment (Ty3uK) [reply lua/null-ls/builtins/diagnostics/swiftlint.lua:36]: No reason at all, sorry.
It's just copy-pasted from `eslint` source 🙈

File: lua/null-ls/builtins/diagnostics/swiftlint.lua (hunk 2 above)
Comment (Ty3uK) [reply lua/null-ls/builtins/diagnostics/swiftlint.lua:37]: By "this" you mean `use_cache`?

Comment (jose-elias-alvarez) [reply lua/null-ls/builtins/diagnostics/swiftlint.lua]: Yes, that's what I meant (GitHub unfortunately doesn't make this totally clear).

**Comment by jose-elias-alvarez** [comment]
Thanks!

**Pull request merged**